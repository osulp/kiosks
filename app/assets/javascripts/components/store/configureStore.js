import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const createStoreWithMiddleware = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore);

export default function configureStore(initialState) {
  return createStoreWithMiddleware(rootReducer, initialState);
}