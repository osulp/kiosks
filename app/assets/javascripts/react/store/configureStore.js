import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

export const createStoreWithMiddleware = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore);

const configureStore = (initialState) => {
  return createStoreWithMiddleware(rootReducer, initialState);
};

export default configureStore;