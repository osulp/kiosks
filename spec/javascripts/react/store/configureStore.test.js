import { default as configureStore } from "../../../../app/javascript/react/store/configureStore"

const empty_state = undefined

describe("Store::configureStore", () => {
  it("matches the snapshot", () => {
    expect(configureStore).toMatchSnapshot()
  })
  it("creates a default store with middleware", () => {
    expect(configureStore(empty_state)).toMatchSnapshot()
  })
})
