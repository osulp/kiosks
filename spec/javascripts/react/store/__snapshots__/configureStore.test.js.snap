// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store::configureStore creates a default store with middleware 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;

exports[`Store::configureStore matches the snapshot 1`] = `[Function]`;
